modules/jolt_physics/shapes/jolt_tapered_capsule_shape_3d.cpp

```diff
@@ -0,0 +1,97 @@
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "jolt_tapered_capsule_shape_3d.h"
+
+#include "../misc/jolt_type_conversions.h"
+
+#include "Jolt/Physics/Collision/Shape/TaperedCapsuleShape.h"
+
+JPH::ShapeRefC JoltTaperedCapsuleShape3D::_build() const {
+	ERR_FAIL_COND_V_MSG(radius1 <= 0.0f, nullptr, vformat("Failed to build Jolt Physics tapered capsule shape with %s. Its radius1 must be greater than 0. This shape belongs to %s.", to_string(), _owners_to_string()));
+	ERR_FAIL_COND_V_MSG(radius2 <= 0.0f, nullptr, vformat("Failed to build Jolt Physics tapered capsule shape with %s. Its radius2 must be greater than 0. This shape belongs to %s.", to_string(), _owners_to_string()));
+	ERR_FAIL_COND_V_MSG(height <= 0.0f, nullptr, vformat("Failed to build Jolt Physics tapered capsule shape with %s. Its height must be greater than 0. This shape belongs to %s.", to_string(), _owners_to_string()));
+	ERR_FAIL_COND_V_MSG(height < radius1 + radius2, nullptr, vformat("Failed to build Jolt Physics tapered capsule shape with %s. Its height must be at least the sum of its radii. This shape belongs to %s.", to_string(), _owners_to_string()));
+
+	const float half_height = height / 2.0f;
+
+	const JPH::TaperedCapsuleShapeSettings shape_settings(half_height, radius1, radius2);
+	const JPH::ShapeSettings::ShapeResult shape_result = shape_settings.Create();
+	ERR_FAIL_COND_V_MSG(shape_result.HasError(), nullptr, vformat("Failed to build Jolt Physics tapered capsule shape with %s. It returned the following error: '%s'. This shape belongs to %s.", to_string(), to_godot(shape_result.GetError()), _owners_to_string()));
+
+	return shape_result.Get();
+}
+
+Variant JoltTaperedCapsuleShape3D::get_data() const {
+	Dictionary data;
+	data["radius1"] = radius1;
+	data["radius2"] = radius2;
+	data["height"] = height;
+	return data;
+}
+
+void JoltTaperedCapsuleShape3D::set_data(const Variant &p_data) {
+	ERR_FAIL_COND(p_data.get_type() != Variant::DICTIONARY);
+
+	const Dictionary data = p_data;
+
+	const Variant maybe_radius1 = data.get("radius1", Variant());
+	ERR_FAIL_COND(maybe_radius1.get_type() != Variant::FLOAT);
+
+	const Variant maybe_radius2 = data.get("radius2", Variant());
+	ERR_FAIL_COND(maybe_radius2.get_type() != Variant::FLOAT);
```

Comment:
Need to check if we are using radius1, radius2 from the jolt api or the godot api of radius_top and radius_bottom.

Reply:
I still think radius_top and radius_bottom is better than radius1 and raidus2